dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/dataitem/helper/OrderingHelper.java

```diff
@@ -0,0 +1,106 @@
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import static org.apache.commons.collections4.CollectionUtils.isNotEmpty;
+import static org.apache.commons.lang3.StringUtils.split;
+import static org.hisp.dhis.feedback.ErrorCode.E2015;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
+
+import org.apache.commons.beanutils.BeanComparator;
+import org.apache.commons.collections4.comparators.ComparatorChain;
+import org.apache.commons.collections4.comparators.NullComparator;
+import org.hisp.dhis.common.BaseDimensionalItemObject;
+import org.hisp.dhis.common.IllegalQueryException;
+import org.hisp.dhis.dxf2.common.OrderParams;
+import org.hisp.dhis.feedback.ErrorMessage;
+
+/**
+ * Helper class responsible for providing sorting capabilities.
+ */
+public class OrderingHelper
+{
+    private static final int ORDERING_ATTRIBUTE = 0;
+
+    private static final int ORDERING_VALUE = 1;
+
+    private static final String DESC = "desc";
+
+    /**
+     * Sorts the given list based on the given sorting params.
+     *
+     * @param dimensionalItems
+     * @param sortingParams
+     */
+    public static void sort( final List<BaseDimensionalItemObject> dimensionalItems, final OrderParams sortingParams )
+    {
+        if ( sortingParams != null && isNotEmpty( dimensionalItems ) )
+        {
+            final ComparatorChain<BaseDimensionalItemObject> chainOfComparators = new ComparatorChain();
+            final Set<String> orderingPairs = sortingParams.getOrders();
+
+            if ( sortingParams != null && isNotEmpty( orderingPairs ) )
+            {
+                for ( final String orderingPair : orderingPairs )
+                {
+                    chainOfComparators.addComparator( getComparator( orderingPair ) );
+                }
+
+                Collections.sort( dimensionalItems, chainOfComparators );
+            }
+        }
+    }
+
+    private static Comparator getComparator( final String orderingParam )
+    {
+        final String[] orderingAttributes = split( orderingParam, ":" );
+        final boolean hasValidOrderingAttributes = orderingAttributes != null & orderingAttributes.length == 2;
+
+        if ( hasValidOrderingAttributes )
+        {
+            final BeanComparator<BaseDimensionalItemObject> comparator = new BeanComparator(
```

Comment:
very minory:
`BeanComparator<>`
`NullComparator<>`
to remove the generics warning

Reply:
Done